### PR TITLE
Context redirect sayani

### PR DESCRIPTION
**Pull request by sayanidutta2345**
>  Check number in the user database and change the output context accordingly.

>  Show profile details of the user when requested

> More info intent

### CODE REVIEW

File: webhook/index.js [hunk 1]
```diff
@@ -33,6 +66,45 @@ router.post('/api/chat/', async function(req, res) {
         );
         responseText = responseText + "\n" + doctors.join("\n");
     }
+    const isUser=function(id){
```
[the remark this answers is not in the document]
Comment (kj7kunal) [reply webhook/index.js:69]: By the way this function is not needed. Instead you can set a variable called `isUser` in the start whenever the twilio request comes in.

File: webhook/index.js [hunk 2]
```diff
@@ -23,6 +23,39 @@ router.post('/api/chat/', async function(req, res) {
...
+        }).then((result) => {
+            const hbsObject = {
+                    User: result
+                }
+                if (result==null){
+                    dialogflowResponse.outputContext[0].name="projects/e-medicine-iitkgp-mvttlt/agent/sessions/"+id +"/contexts/not_reg";
+                    dialogflowResponse.outputContext[0].lifespanCount= 5;
+                    //dialogflowResponse.outputContext.parameters=[Object];
+                }
+                else if(result.dataValues.type =='doctor'){
```
Comment (kj7kunal) [inline webhook/index.js:41]: There is no `type` in a `userWA` object

File: webhook/handlers/test/sayani.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
+'use strict'
```
[the remark this answers is not in the document]
Comment (sayanidutta2345) [reply webhook/handlers/test/sayani.js:1]: Yeahh got it

**Comment by sayanidutta2345** [comment]
Indent corrected
userWAs->userWA
userWA type is not there so just created a general output indent for registered users
Extra spacing removed
LinkedIn link changed
id substring for wa_phone_number

File: package-lock.json [hunk 4]
```diff
@@ -8,24 +8,21 @@
...
             "integrity": "sha512-ihn9xWOqubMPBlU77wcYpy7FFamGo5xtsK27EAILL/eoOvGEAq29UOrqRvqYPwWfl2+3laFmGKNR7uCdJhKu4Q==",
-            "dev": true,
             "requires": {
                 "semver": "^6.2.0"
             },
             "dependencies": {
                 "semver": {
                     "version": "6.3.0",
                     "resolved": "https://registry.npmjs.org/semver/-/semver-6.3.0.tgz",
-                    "integrity": "sha512-b39TBaTSfV6yBrapU89p5fKekE2m/NwnDocOVruQFS1/veMgdzuPcnOM34M6CwxW8jH/lxEa5rBoDeUwu5HHTw==",
```
[the remark this answers is not in the document]
Comment (sayanidutta2345) [reply package-lock.json:19]: okay

File: webhook/index.js [hunk 5]
```diff
@@ -23,6 +23,39 @@ router.post('/api/chat/', async function(req, res) {
 
     const dialogflowResponse = (await sessionClient.detectIntent(
         text, id, body)); // Gets intent
+        //dialogflowResponse=check_num_indb(dialogflowResponse,id);
+        
+        db.userWAs.findOne({
```
[the remark this answers is not in the document]
Comment (sayanidutta2345) [reply webhook/index.js]: done

File: webhook/index.js [hunk 6]
```diff
@@ -23,6 +23,39 @@ router.post('/api/chat/', async function(req, res) {
...
+            where: {
+                wa_phone_number: id
+            }
+        }).then((result) => {
```
[the remark this answers is not in the document]
Comment (sayanidutta2345) [reply webhook/index.js]: done

**Pull request merged**